tests/unittests/utils/test_systemd.py

```diff
@@ -0,0 +1,303 @@
+# -*- coding: utf-8 -*-
+
+# Import python libs
+from __future__ import absolute_import, print_function, unicode_literals
+import errno
+import os
+
+# Import Salt Testing libs
+from tests.support.unit import TestCase, skipIf
+from tests.support.mock import Mock, patch, NO_MOCK, NO_MOCK_REASON
+
+# Import Salt libs
+import hubblestack.utils.systemd as _systemd
+from hubblestack.utils.exceptions import HubbleInvocationError
+
+
+def _booted_effect(path):
+    return True if path == '/run/systemd/system' else os.stat(path)
+
+
+def _not_booted_effect(path):
+    if path == '/run/systemd/system':
+        raise OSError(errno.ENOENT, 'No such file or directory', path)
+    return os.stat(path)
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
+class SystemdTestCase(TestCase):
+    '''
+    Tests the functions in salt.utils.systemd
+    '''
+    def test_booted(self):
+        '''
+        Test that salt.utils.systemd.booted() returns True when minion is
+        systemd-booted.
+        '''
+        # Ensure that os.stat returns True. os.stat doesn't return a bool
+        # normally, but the code is doing a simple truth check on the return
+        # data, so it is sufficient enough to mock it as True for these tests.
+        with patch('os.stat', side_effect=_booted_effect):
+            # Test without context dict passed
+            self.assertTrue(_systemd.booted())
+            # Test that context key is set when context dict is passed
+            context = {}
+            self.assertTrue(_systemd.booted(context))
+            self.assertEqual(context, {'salt.utils.systemd.booted': True})
+
+    def test_not_booted(self):
+        '''
+        Test that salt.utils.systemd.booted() returns False when minion is not
+        systemd-booted.
+        '''
+        # Ensure that os.stat raises an exception even if test is being run on
+        # a systemd-booted host.
+        with patch('os.stat', side_effect=_not_booted_effect):
+            # Test without context dict passed
+            self.assertFalse(_systemd.booted())
+            # Test that context key is set when context dict is passed
+            context = {}
+            self.assertFalse(_systemd.booted(context))
+            self.assertEqual(context, {'salt.utils.systemd.booted': False})
```

Comment:
Change salt to hubblestack
Change at all places in this file

Reply:
fixed